src/main/java/com/flowingcode/vaadin/addons/cronexpressioneditor/CronExpressionEditor.java

```diff
@@ -0,0 +1,595 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+package com.flowingcode.vaadin.addons.cronexpressioneditor;
+
+import java.time.DayOfWeek;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+import com.flowingcode.vaadin.addons.dayofweekselector.DayOfWeekSelector;
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.button.ButtonVariant;
+import com.vaadin.flow.component.combobox.ComboBox;
+import com.vaadin.flow.component.customfield.CustomField;
+import com.vaadin.flow.component.dependency.CssImport;
+import com.vaadin.flow.component.formlayout.FormLayout;
+import com.vaadin.flow.component.formlayout.FormLayout.ResponsiveStep;
+import com.vaadin.flow.component.grid.Grid;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.Paragraph;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.orderedlayout.FlexComponent.Alignment;
+import com.vaadin.flow.component.orderedlayout.FlexComponent.JustifyContentMode;
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.component.textfield.IntegerField;
+import com.vaadin.flow.component.textfield.TextField;
+import com.vaadin.flow.component.timepicker.TimePicker;
+import com.vaadin.flow.data.provider.DataProvider;
+import com.vaadin.flow.function.SerializableConsumer;
+import com.vaadin.flow.internal.JsonSerializer;
+import com.vaadin.flow.theme.lumo.LumoUtility;
+import elemental.json.JsonObject;
+import org.springframework.format.annotation.DurationFormat.Unit;
+import org.springframework.scheduling.support.CronExpression;
+import it.burning.cron.CronExpressionDescriptor;
+import it.burning.cron.CronExpressionParser.CronExpressionParseException;
+import it.burning.cron.CronExpressionParser.Options;
+
+
+/**
+ * UI component for building cron expressions.
+ *
+ * @author Sofia Nuñez / Flowing Code
+ */
+@SuppressWarnings("serial")
+@CssImport(value = "./styles/cron-expression-editor-styles.css")
+public class CronExpressionEditor extends CustomField<String> {
+
+  /**
+   * Default value used only when {@code inputExpressionTf} and {@code defaultExpression} are empty. Represents every
+   * day at midnight.
+   */
+  private static final String DEFAULT_CRON = "0 0 0 * * *";
+  private Locale locale;
```

Comment:
Consider removing `locale` and base `CronExpressionDescriptor` on `getLocale()` (inherited from `Component`)
```suggestion
```